docs/DeveloperGuide.md

```diff
@@ -426,6 +438,95 @@ In the diagram, the predicates `modulePredicate` and `taskPredicate` are the cus
 
 <img src="images/CdSequenceDiagram.png" width="1000" />
 
+### Marking modules as done
+
+Section by : [Marciano](https://github.com/midnightfeverrr)
+
+Marking modules as done allows users to selectively hide modules that they have completed.
+
+Every module in the module list will be in either the done or undone state.
+New tag/label will be created to mark a module as `done`.
+
+In this section, we will discuss the management of done/undone state, as well as the
+interactions between the commands and the UI.
+
+The relevant commands for this section are:
+* **`done module <module code>`**  marks the module visible in the UI with the specified module code as done.
```

Comment:
again, this is a bit ambiguous as the module is always visible, perhaps this could be refactored to
`marks the module with a done tag in the UI` and `removes the done tag from a module (if present) in the UI` for the undone command